blaze-client/src/main/scala/org/http4s/blaze/client/Http1Connection.scala

```diff
@@ -310,105 +310,113 @@ private final class Http1Connection[F[_]](
         cb(Left(t))
     }(executionContext)
 
-  private def parsePrelude(
+  // it's called when the body is known
+  private def terminationConditionParsingPrelude(): Either[Throwable, Option[Chunk[Byte]]] =
+    stageState.get match { // if we don't have a length, EOF signals the end of the body.
+      case Error(e) if e != EOF => Either.left(e)
+      case _ =>
+        if (parser.definedContentLength() || parser.isChunked())
+          Either.left(InvalidBodyException("Received premature EOF."))
+        else Either.right(None)
+    }
+
+  private def cleanupParsingPrelude(closeOnFinish: Boolean, headers: Headers): Unit =
```

Comment:
Maybe `cleanupParsingResponse`? Similarly to my comment on `terminationConditionParsingPrelude`. The `cleanupParsingPrelude` happens later than after parsing prelude. First we parse the prelude, then the body, then (optionally) the trailers and then we clean up.

IMO it should be `cleanUp` (a verb) not `cleanup` (a noun).